File: java/src/Rosliny/Guarana.java
package Rosliny;
import Swiat.*;

public class Guarana extends Roslina{
    public Guarana(int x, int y, Swiat swiat){
        super(Gatunek.GUARANA, 0, 0, x, y, swiat);
    }
    public Guarana(int x, int y, int sila, Swiat swiat){
        super(Gatunek.GUARANA, sila, 0, x, y, swiat);
    }

    @Override
    public void kolizja(Organizm organizm){
        this.walka(organizm);
        organizm.zwiekszSila(3);
    }

    @Override
    public String rysowanie(){return "G";}
}

File: python/plants/guarana.py
from plants.roslina import Roslina


class Guarana(Roslina):

    def __init__(self, x, y, world, sila=0):
        Roslina.__init__(self, "guarana", sila, 0, x, y, world)

    def __str__(self):
        return self._species + "[" + str(self._x) + "][" + str(self._y) + "]"

    def kolizja(self, o):
        self._world.addLog(o.__str__() + " je " + self.__str__())
        self.kill()
        o.setPolozenie(self._x, self._y)
        o.addSila(3)

    def rysowanie(self):
        return "G"
